crates/matrix-sdk-ui/src/timeline/event_type_filter.rs

```diff
@@ -0,0 +1,21 @@
+use ruma::events::{AnySyncTimelineEvent, TimelineEventType};
+
+/// A timeline filter that either includes only events with event_type included
+/// in a list or all but a list of excluded ones
+pub enum TimelineEventTypeFilter {
+    Include(Vec<TimelineEventType>),
+    Exclude(Vec<TimelineEventType>),
+}
+
+impl TimelineEventTypeFilter {
+    pub fn filter(&self, event: &AnySyncTimelineEvent) -> bool {
```

Comment:
Can you add a doc comment please? (does Clippy not complain about the missing doc-comment here :eyes:)

Reply:
Will do, and no, clippy didn't seem to complain, that's weird.

Reply:
I think we only enable the `missing_docs` lint on select crates.

Reply:
Please doc comment here :eyes: :grin: